File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        TreeNode left1 = null;
        TreeNode right1 = null;
        if (root == null){
            return root;
        }
        else{
            left1 = root.left;
            right1 = root.right;
            root.right = left1;
            root.left = right1;
            invertTree(root.left);
            invertTree(root.right);
    }
    return root;
}
}

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.answer = []

        def dfs(root):
            stack = [root]
            visited = [False]

            if not root:
                return self.answer


            while stack:
                curr, seen = stack.pop(), visited.pop()
                if curr:
                    if seen:
                        self.answer.append(curr.val)

                    else:
                        stack.append(curr); visited.append(True)
                        stack.append(curr.right); visited.append(False)
                        stack.append(curr.left); visited.append(False)

        dfs(root)

        return self.answer